rclcpp/resource/logging.hpp.em

```diff
@@ -149,10 +160,12 @@ def get_rclcpp_suffix_from_features(features):
 @(''.join(['      ' + p + ', \\\n' for p in params if p != stream_arg]))@
 @[ end if]@
       (logger).get_name(), \
-@[ if 'stream' not in feature_combination]@
-      __VA_ARGS__); \
-@[ else]@
+@[ if 'stream' in feature_combination]@
       "%s", rclcpp_stream_ss_.str().c_str()); \
+@[ elif 'format' in feature_combination]@
+      "%s", std::format(__VA_ARGS__).c_str()); \
```

Comment:
this could possibly introduce `std::format_error`?

Reply:
Yes, `std::format` could throw `std::format_error` when the given format string is invalid for the provided arguments. However, in C++23 with [P2216](http://wg21.link/p2216), this is replaced by compile-time checks.